lib/cocoapods/target.rb

```diff
@@ -41,23 +34,21 @@ class Target
     #
     attr_reader :build_settings
 
-    # @return [Target::BuildType] the build type for this target.
+    # @return [BuildType] the build type for this target.
     #
     attr_reader :build_type
     private :build_type
 
     # Initialize a new target
     #
     # @param [Sandbox] sandbox @see #sandbox
-    # @param [Boolean] host_requires_frameworks @see #host_requires_frameworks
+    # @param [BuildType] build_type @see #build_type
     # @param [Hash{String=>Symbol}] user_build_configurations @see #user_build_configurations
     # @param [Array<String>] archs @see #archs
     # @param [Platform] platform @see #platform
     #
-    def initialize(sandbox, host_requires_frameworks, user_build_configurations, archs, platform,
-                   build_type: Target::BuildType.infer_from_spec(nil, :host_requires_frameworks => host_requires_frameworks?))
+    def initialize(sandbox, build_type, user_build_configurations, archs, platform)
       @sandbox = sandbox
-      @host_requires_frameworks = host_requires_frameworks
```

Comment:
the change to arguments is backwards incompatible, so it would break any plugins that create instances of `PodTarget` (ex. `cocoapods-packager`)

should we check `if [TrueClass, FalseClass].include?(build_type.class)` or something similar and emit a warning? maybe we can infer a default until 2.x

Reply:
thanks! good point and good to know. Will adjust accordingly.

Reply:
@amorde are you suggesting to keep the `attr` of `host_requires_frameworks?` in the initializer parameters or just provide a method of it so we maintain backwards compatibility?

Reply:
@amorde going to restructure this PR into 2 commits. I mistakenly did it in one single commit and we will see if I keep the commit that removes `host_requires_frameworks` part.

Reply:
@amorde spoke briefly with @segiddins offline and it seems that this "breaking" change might be "OK". I'll let him if he finds time to respond to this though 100%.